integreat_cms/cms/views/organizations/organizations_form_view.py

```diff
@@ -0,0 +1,127 @@
+from django.views.generic import TemplateView
+
+
+from ...constants import status, translation_status
+from ...decorators import permission_required
+from ...forms import OrganizationForm
+from ..media.media_context_mixin import MediaContextMixin
+from ..mixins import ContentEditLockMixin
+
+
+logger = logging.getLogger(__name__)
+
+
+@method_decorator(permission_required("cms.view_organization"), name="dispatch")
+# @method_decorator(permission_required("cms.change_event"), name="post")
+class OrganizationFromView(TemplateView, MediaContextMixin, ContentEditLockMixin):
+    """
+    Class for rendering the events form
+    """
+
+    #: The template to render (see :class:`~django.views.generic.base.TemplateResponseMixin`)
+    template_name = "organizations/organization_form.html"
+    #: The context dict passed to the template (see :class:`~django.views.generic.base.ContextMixin`)
+    extra_context = {
+        "current_menu_item": "organizations",
+        "PUBLIC": status.PUBLIC,
+        "translation_status": translation_status,
+    }
+    #: The url name of the view to show if the user decides to go back (see :class:`~integreat_cms.cms.views.mixins.ContentEditLockMixin`
+    back_url_name = "organizations"
+
+    # pylint: disable=too-many-locals
+    def get(self, request, *args, **kwargs):
+        r"""
+        Render event form for HTTP GET requests
+
+        :param request: Object representing the user call
+        :type request: ~django.http.HttpRequest
+
+        :param \*args: The supplied arguments
+        :type \*args: list
+
+        :param \**kwargs: The supplied keyword arguments
+        :type \**kwargs: dict
+
+        :return: The rendered template response
+        :rtype: ~django.template.response.TemplateResponse
+        """
+        region = request.region
+
+        # get event and event translation objects if they exist, otherwise objects are None
+        organization_instance = region.organizations.filter(
+            slug=kwargs.get("slug")
+        ).first()
+
+        # Make form disabled if event is archived or user doesn't have the permission to edit the event
+        if not request.user.has_perm("cms.change_organization"):
+            disabled = True
+            messages.warning(
+                request, _("You don't have the permission to edit an organization.")
+            )
+        else:
+            disabled = False
+
+        organization_form = OrganizationForm(instance=organization_instance)
+        logger.warning(self.get_context_data(**kwargs))
+        return render(
+            request,
+            self.template_name,
+            {
+                **self.get_context_data(**kwargs),
+                "form": organization_form,
+            },
+        )
+
+    def post(self, request, *args, **kwargs):
+        region = request.region
+        organization_instance = region.organizations.filter(
+            slug=kwargs.get("slug")
+        ).first()
```

Comment:
lgtm. Although, I think the situations were this might occur are very limited (maybe if the organization is deleted by another user after we already rendered the page?)

Reply:
Yes, or when somebody has bookmarked the edit form and opens it after the orga has been deleted (or navigates back in the browser a few times too often)... but of course these are only edge cases...